pkg/asset/installconfig/baremetal/baremetal.go

```diff
@@ -2,164 +2,10 @@
-			},
-			Validate: survey.ComposeValidators(survey.Required, interfaceValidator),
-		},
-	}, &externalBridge)
-	if err != nil {
-		return nil, err
-	}
-
-	err = survey.Ask([]*survey.Question{
-		{
-			Prompt: &survey.Input{
-				Message: "Provisioning bridge",
-				Help:    "Provisioning bridge is used to provision machines.",
-				Default: baremetaldefaults.ProvisioningBridge,
-			},
-			Validate: survey.ComposeValidators(survey.Required, interfaceValidator),
-		},
-	}, &provisioningBridge)
-	if err != nil {
-		return nil, err
-	}
-
-	err = survey.Ask([]*survey.Question{
-		{
-			Prompt: &survey.Input{
-				Message: "API VIP",
-				Help:    "The VIP to be used for internal API communication. If blank, the address will be looked up from DNS.",
-				Default: baremetaldefaults.APIVIP,
-			},
-			Validate: apiVIPValidator,
-		},
-	}, &apiVIP)
-	if err != nil {
-		return nil, err
-	}
-
-	err = survey.Ask([]*survey.Question{
-		{
-			Prompt: &survey.Input{
-				Message: "Ingress VIP",
-				Help:    "The VIP to be used for ingress. If blank, the address will be looked up from DNS.",
-				Default: baremetaldefaults.IngressVIP,
-			},
-			Validate: ingressVIPValidator,
-		},
-	}, &ingressVIP)
-	if err != nil {
-		return nil, err
-	}
-
-	// Keep prompting for hosts
-	for {
-		var host *baremetal.Host
-		host, err = Host()
-		host.Role = "master" // FIXME(stbenjam): Support workers
-
-		if err != nil {
-			fmt.Printf("Invalid host - please try again")
-			continue
-		}
-		hosts = append(hosts, host)
-
-		more := false
-		prompt := &survey.Confirm{
-			Message: "Add another?",
-		}
-		survey.AskOne(prompt, &more, nil)
-		if !more {
-			break
-		}
-	}
-
-	return &baremetal.Platform{
-		LibvirtURI:         libvirtURI,
-		IronicURI:          ironicURI,
-		APIVIP:             apiVIP,
-		IngressVIP:         ingressVIP,
-		ExternalBridge:     externalBridge,
-		ProvisioningBridge: provisioningBridge,
-		Hosts:              hosts,
-	}, nil
-}
-
-// uriValidator validates if the answer provided in prompt is a valid
-// url and has non-empty scheme.
-func uriValidator(ans interface{}) error {
-	return validate.URI(ans.(string))
-}
-
-func apiVIPValidator(ans interface{}) error {
-	if (ans.(string) != baremetaldefaults.APIVIP) {
-		return validate.IP(ans.(string))
-	}
-	return nil
-}
-
-func ingressVIPValidator(ans interface{}) error {
-	if (ans.(string) != baremetaldefaults.IngressVIP) {
-		return validate.IP(ans.(string))
-	}
-	return nil
-}
-
-// interfaceValidator validates if the answer provided is a valid network
-// interface. net.Interfaces in Go does not let us know if it's a bridge, but
-// we can at least make sure an interface by that name exists.
-func interfaceValidator(ans interface{}) error {
-	return validate.Interface(ans.(string))
-}
-
-// macValidator validates if the answer provided is a valid mac address
-func macValidator(ans interface{}) error {
-	return validate.MAC(ans.(string))
+	return &baremetal.Platform{}, nil
```

Comment:
This code is unreachable since the platform is hidden anyway, we could really delete it entirely.  I'm not sure what the best approach is. If we'll ever support the TUI, maybe it makes sense to leave it stubbed.

I could see if we ever use discovery the TUI becomes more usable, as a user would just be able to select which bare metal hosts to use for the control plane, for example.

Reply:
If we don't validate install-config data ... that's fascinating

In general, +1 to just deleting unused code even if we might want to bring it back in future